src/operators/Cylinder/PyCylinderAttributes.C

```diff
@@ -156,12 +156,12 @@ CylinderAttributes_SetPoint2(PyObject *self, PyObject *args)
     {
         PyObject     *tuple;
         if(!PyArg_ParseTuple(args, "O", &tuple))
-            return NULL;
+            return PyExc_TypeError;
 
         if(PyTuple_Check(tuple))
         {
             if(PyTuple_Size(tuple) != 3)
-                return NULL;
+                return PyExc_ValueError;
```

Comment:
I noticed that AttVector sets an error message for its IndexError exception.
Would it make sense to use that same functionality for this ValueError exception so that we can indicate that the tuple is the wrong size? Perhaps even use the same kind of messaging for other exceptions where the generic error message in the setattr could possibly be misleading or just not be providing enough information?

https://github.com/visit-dav/visit/blob/1712b0f85c167aa19ff0a7a97dde2fc13681d03c/src/tools/dev/xml/GeneratePython.h#L2148-L2152

Reply:
Do you think its worthwhile to do that for all the other (~10) instances where we have similar logic?

Reply:
Also, if we call `PyErr_SetString(PyExc_IndexError, ...)` and return `NULL` here, what happens to that string when the caller catches the `NULL` return and then turns around and calls `PyErr_Format(PyExc_RunTimeError,...)` with a different error object and a different string? Are the first values lost?